banking_system/exceptions/exception.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+namespace exceptions {
+    class Exception: public std::exception {
+    public:
+        explicit Exception() :m_message( "Exception raised! Details: ") {}
+        char const* what() const noexcept override { return m_message.c_str(); }
+    protected:
+        std::string m_message;
```

Comment:
We can avoid to use protected field here, just use constructor with parameter, that takes error message and put it into field, and all derived classes can call in initialisation list just Exception(message)